SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -277,6 +290,8 @@ void ASpatialFunctionalTest::FinishTest(EFunctionalTestResult TestResult, const
 		// Make sure we don't FinishTest multiple times.
 		if (CurrentStepIndex != SPATIAL_FUNCTIONAL_TEST_FINISHED)
 		{
+			bPreparedTest = false; // Clear for PrepareTest to run on all again if the test re-runs.
```

Comment:
Is there any possibility that FinishTest could get called in the same frame that PrepareTest is called? In which case this bool wouldn't be replicated.

Reply:
I don't think so. The Authority will get PrepareTest called, then it needs to make sure that IsReady is working, then StartTest will call StartStep(0) and tell whomever is doing it to do it. Then whomever needs to execute will need to call FinishStep() which will trigger a FinishTest which will ask for confirmation on the FinishTest. I highly doubt that this whole thing can happen in the same tick, but can't confirm. The only thing we can do is that each FlowCotnroller to only be ready for the test when they also acknowledge that the bPrepareTest is true on them.
What do you think? Change it to be like that?

Reply:
> FlowCotnroller to only be ready for the test when they also acknowledge that the bPrepareTest is true on them.
What do you think? Change it to be like that?
Sounds good - how do flow controllers currently ready up?

Reply:
There's a flag in the test which gets replicated, and then all of the workers on rep notify mark their controller as being ready. This in turn gets replicated, and the test authority starts the test when all of them are locally ready.

Reply:
So workers would now block marking their flow controller ready until they've seen the prepared test flag?
It does sound like there are two flags now that are essentially doing the same thing.

Reply:
Yes, if we change it it would work like that. However, it may not be needed like I wrote in private chat.